src/components/ResultsGrid.tsx

```diff
@@ -125,67 +137,110 @@ export default function ResultsGrid() {
-            Stream
-          </Column>
-          <Column id={`${config.fieldNames.DWR_WaterID}_Stream`} minWidth={120}>
-            Stream ID
-          </Column>
-          <Column id={`${config.fieldNames.ReachCode}_Stream`} minWidth={200}>
-            Stream Reach Code
-          </Column>
-          <Column id={`${config.fieldNames.WaterName}_Lake`} minWidth={150}>
-            Lake
-          </Column>
-          <Column id={`${config.fieldNames.DWR_WaterID}_Lake`} minWidth={120}>
-            Lake ID
-          </Column>
-          <Column id={`${config.fieldNames.ReachCode}_Lake`} minWidth={200}>
-            Lake Reach Code
-          </Column>
-          <Column id={STATION_NAME} minWidth={180}>
-            Station Name
-          </Column>
-          <Column id={config.fieldNames.SPECIES} minWidth={180}>
-            Species Codes
-          </Column>
-          <Column id={config.fieldNames.TYPES} minWidth={150}>
-            Equipment
-          </Column>
-          <Column id={config.fieldNames.EVENT_ID} isRowHeader minWidth={350}>
-            Event ID
-          </Column>
-        </TableHeader>
-        <TableBody items={data}>
-          {(row) => (
-            <Row>
-              <Cell>{new Date(row[config.fieldNames.EVENT_DATE] as number).toLocaleDateString()}</Cell>
-              <Cell>{row[config.fieldNames.OBSERVERS]}</Cell>
-              <Cell>{row[`${config.fieldNames.WaterName}_Stream`]}</Cell>
-              <Cell>{row[`${config.fieldNames.DWR_WaterID}_Stream`]}</Cell>
-              <Cell>{row[`${config.fieldNames.ReachCode}_Stream`]}</Cell>
-              <Cell>{row[`${config.fieldNames.WaterName}_Lake`]}</Cell>
-              <Cell>{row[`${config.fieldNames.DWR_WaterID}_Lake`]}</Cell>
-              <Cell>{row[`${config.fieldNames.ReachCode}_Lake`]}</Cell>
-              <Cell>{row[STATION_NAME]}</Cell>
-              <Cell>{row[config.fieldNames.SPECIES]}</Cell>
-              <Cell>{row[config.fieldNames.TYPES]}</Cell>
-              <Cell>{row[config.fieldNames.EVENT_ID]}</Cell>
-            </Row>
-          )}
-        </TableBody>
-      </Table>
+      <span className="absolute right-12 top-2 z-10 self-center">
+        Records: <strong>{data?.length}</strong>
+        {selectedStationIds.size > 0 && (
+          <span>
+            {' '}
+            | Selected: <strong>{selectedKeys === 'all' ? data?.length : selectedKeys.size}</strong>
+            <Button
+              variant="secondary"
+              // size="extraSmall"
+              onPress={() => setSelectedStationIds(new Set())}
+              className="ml-2"
+            >
+              Clear Selection
+            </Button>
+          </span>
+        )}
+      </span>
+      <Tabs aria-label="results panel">
+        <TabList>
+          <Tab id="grid">Results</Tab>
+          <Tab id="download">Download</Tab>
+        </TabList>
+        <TabPanel id="grid">
+          <Table
```

Comment:
**thought**: I believe every time the tabs are switched the content is reloaded and any state is lost. 

Can you explain the steps to identify the performance issue you talk about?

Would the tab state be related?

Reply:
I'm not sure why there is a performance issue. It's much worse in development than in the production build. I just added a story to demonstrate the issue and show that it's specific to the table component.

Reply:
I've tried profiling in dev tools and ran into this bug: https://github.com/facebook/react/issues/25415

Reply:
I'm not fixed to this table implementation. With the amount of experience we have with tanstack table, I'd be happy to use that if you think it makes more sense.

Reply:
https://github.com/adobe/react-spectrum/issues/5707